core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1067,17 +1065,26 @@ public FileSystemMasterView getFileSystemMasterView() {
       // Possible ufs sync.
       syncMetadata(rpcContext, parent, lockingScheme, DescendantType.ALL);
 
-      try (InodeLockList children = mInodeTree.lockDescendants(parent, InodeTree.LockMode.READ)) {
-        if (!checkConsistencyInternal(parent.getInode(), parent.getUri())) {
-          inconsistentUris.add(parent.getUri());
+      if (!checkConsistencyInternal(parent.getInode(), parent.getUri())) {
+        inconsistentUris.add(parent.getUri());
+      }
+
+      List<LockedInodePath> children = mInodeTree.lockDescendants(parent, InodeTree.LockMode.READ);
```

Comment:
This pattern is found in several places. Maybe we should have a new class which is closeable, so we can just use try-with-resources on the list.

Reply:
are you suggesting we make a `LockedInodePathList` class that is closeable?

Reply:
Yeah, maybe? Would that simplify the usage of `lockDescendants`?